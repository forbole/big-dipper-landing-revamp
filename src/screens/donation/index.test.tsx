import React from 'react';
import { render, screen } from '@testing-library/react';
import { MockTheme } from '@mocks';
import Component from '.';

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Layout: (props: any) => <div data-testid="Layout" {...props} />,
  SectionBox: (props: any) => <div data-testid="SectionBox" {...props} />,
  SectionLimit: (props: any) => <div data-testid="SectionLimit" {...props} />,
  ContentBox: (props: any) => <div data-testid="ContentBox" {...props} />,
}));
// ==================================
// unit tests
// ==================================
describe('Donation', () => {
  it('matches snapshot', async () => {
    const { container } = render(
      <MockTheme>
        <Component />
      </MockTheme>,
    );

    expect(screen.getByText(
      /donation/i,
    )).toBeInTheDocument();
    expect(screen.getByText(
      /description/i,
    )).toBeInTheDocument();
    expect(screen.getByTestId('Layout')).toBeInTheDocument();
    expect(screen.getByTestId('SectionBox')).toBeInTheDocument();
    expect(screen.getByTestId('SectionLimit')).toBeInTheDocument();
    expect(screen.getAllByTestId('ContentBox').length).toBe(3);
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
