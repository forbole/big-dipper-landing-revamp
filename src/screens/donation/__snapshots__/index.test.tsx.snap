// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donation matches snapshot 1`] = `
.emotion-0 .title {
  margin-bottom: 24px;
  text-align: center;
}

@media (min-width:1280px) {
  .emotion-0 .title {
    margin-bottom: 16px;
    text-align: left;
  }
}

.emotion-0 .description {
  margin-bottom: 32px;
}

.emotion-0 .address__container {
  display: grid;
  grid-gap: 24px;
  text-align: center;
}

@media (min-width:1280px) {
  .emotion-0 .address__container {
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 32px;
  }
}

.emotion-0 .address__logo {
  margin-bottom: 24px;
}

.emotion-0 .address__key {
  margin-bottom: 8px;
}

.emotion-0 .address__address {
  word-break: break-all;
  margin-bottom: 24px;
}

.emotion-0 .address__button--copy {
  margin-right: 8px;
}

.emotion-0 .address__button {
  padding: 0.75rem 1rem;
  border-radius: 200px;
  background: linear-gradient(103.18deg, #F6504B 20.77%, #DC001A 118.69%);
  color: #FFFFFF;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  width: 100%;
}

.emotion-0 .address__button:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.emotion-1 {
  margin: 0;
  font-size: 2rem;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.2;
}

.emotion-2 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  font-family: SFPro,Arial,sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

.emotion-3 {
  margin: 0;
  font-size: 1.25rem;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.235;
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 1rem;
  text-transform: none;
  font-weight: 600;
  font-family: SFPro,Arial,sans-serif;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: rgba(0, 0, 0, 0.87);
  background-color: #90caf9;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  box-shadow: none;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #42a5f5;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: #90caf9;
  }
}

.emotion-5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-5.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
  box-shadow: none;
  background-color: rgba(255, 255, 255, 0.12);
}

.emotion-5:hover {
  box-shadow: none;
}

.emotion-5.Mui-focusVisible {
  box-shadow: none;
}

.emotion-5:active {
  box-shadow: none;
}

.emotion-5.Mui-disabled {
  box-shadow: none;
}

.emotion-6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    data-testid="Layout"
  >
    <div
      data-testid="SectionBox"
    >
      <div
        class="emotion-0"
        data-testid="SectionLimit"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 title emotion-1"
        >
          donation
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 description emotion-2"
        >
          description
        </p>
        <div
          class="address__container"
        >
          <div
            data-testid="ContentBox"
          >
            <img
              alt="btc"
              class="address__logo"
              src="/images/donation/btc.svg"
            />
            <h4
              class="MuiTypography-root MuiTypography-h4 address__key emotion-3"
            >
              btc
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 address__address emotion-2"
            >
              bc1qye4k27zsn5nehzded6jwsvzg8qd6kgvxyhckts
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation address__button emotion-5"
              tabindex="0"
              type="button"
            >
              <span
                class="address__button--copy"
              />
              copy
              <span
                class="MuiTouchRipple-root emotion-6"
              />
            </button>
          </div>
          <div
            data-testid="ContentBox"
          >
            <img
              alt="eth"
              class="address__logo"
              src="/images/donation/eth.svg"
            />
            <h4
              class="MuiTypography-root MuiTypography-h4 address__key emotion-3"
            >
              eth
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 address__address emotion-2"
            >
              0x8CAb9F3fC6bBBD819050365627FC6B79d0ea73e6
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation address__button emotion-5"
              tabindex="0"
              type="button"
            >
              <span
                class="address__button--copy"
              />
              copy
              <span
                class="MuiTouchRipple-root emotion-6"
              />
            </button>
          </div>
          <div
            data-testid="ContentBox"
          >
            <img
              alt="atom"
              class="address__logo"
              src="/images/donation/atom.svg"
            />
            <h4
              class="MuiTypography-root MuiTypography-h4 address__key emotion-3"
            >
              atom
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 address__address emotion-2"
            >
              cosmos1n67vdlaejpj3uzswr9qapeg76zlkusj5k875ma
            </p>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation address__button emotion-5"
              tabindex="0"
              type="button"
            >
              <span
                class="address__button--copy"
              />
              copy
              <span
                class="MuiTouchRipple-root emotion-6"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
